ddl/column.go

```diff
@@ -1354,9 +1407,17 @@ func modifyColsFromNull2NotNull(w *worker, dbInfo *model.DBInfo, tblInfo *model.
 	defer w.sessPool.put(ctx)
 
 	// If there is a null value inserted, it cannot be modified and needs to be rollback.
-	err = checkForNullValue(ctx, isModifiedType, dbInfo.Name, tblInfo.Name, newColName, cols...)
-	if err != nil {
-		return errors.Trace(err)
+	skipCheck := false
+	failpoint.Inject("skipMockContextDoExec", func(val failpoint.Value) {
```

Comment:
Why need the mock `skipMockContextDoExec` ?

Reply:
ddl created with `testNewDDLAndStart` in TestCancelJob doesn't contain the `sessionPool` to get one and run internal SQL